lib/modules/platform/github/index.ts

```diff
@@ -166,14 +166,31 @@ export async function initPlatform({
     }
   }
   logger.debug({ platformConfig, renovateUsername }, 'Platform config');
-  const platformResult: PlatformResult = {
+  const githubDS: HostRule[] = [];
+  const gitHubRule: HostRule[] = [
+    {
+      matchHost: 'ghcr.io',
+      username: renovateUsername,
+      password: token,
+    },
+  ];
+  if (!platformConfig.isGhe) {
+    gitHubRule.push({
+      matchHost: '.pkg.github.com',
+      token,
+    });
+  }
+  for (const rule of gitHubRule) {
+    githubDS.push(rule);
+    hostRules.add(rule);
```

Comment:
i'm not sure why maron added this hostRules, it still works wihtout it, can i remove that?

Reply:
If initPlatform() returns `hostRules`, what happens after that? e.g. where in workers are they added, do they persist over any hostRules clearing, etc?

Reply:
it is added to the config after `const platformInfo = await platform.initPlatform(config);` there is:
`const returnConfig: any = { ...config, ...platformInfo };`

Reply:
We do not need to call `.add()` here, which means this for loop can be removed and logic simplified